src/lib/DownloadControl.ts

```diff
@@ -0,0 +1,60 @@
+import {Map} from 'mapbox-gl'
+import Vue from 'vue'
+
+export class DownloadControl {
+  bbox: any | undefined
+  private _map: Map | undefined
+  private _container: HTMLElement | undefined
+  private _btn: HTMLButtonElement | undefined
+  private _vueComponent: any
+
+  constructor(bbox: any, vueComponent: Vue) {
+    this.bbox = bbox
+    this._vueComponent = vueComponent
+  }
+
+  onAdd(map: Map) {
+    this._map = map
+
+    this._btn = document.createElement('button')
+    this._btn.className = 'mapboxgl-ctrl-download'
+    this._btn.type = 'button'
+
+    const icon = document.createElement('i')
```

Comment:
Why is the icon in a `<i>` tag?

Reply:
I thought that it was the "normal" approach https://www.geeksforgeeks.org/how-to-add-icons-in-the-button-in-html/